### DISCUSSION TITLE
Is directly linking between EnvironmentalMonitoringNetwork and EnvironmentalMonitoringFacility without using the association class NetworkFacility allowed?

### DISCUSSION THREAD

**Discussion by Kate-Lyndegaard** [INSPIRE-MIF/helpdesk#131] [Q&A] [answered]
Hi there,

I have a question regarding the Environmental Monitoring Facility specification. Is directly linking between ```EnvironmentalMonitoringNetwork``` and ```EnvironmentalMonitoringFacility``` without using the association class ```NetworkFacility``` allowed?

UML: https://inspire.ec.europa.eu/data-model/approved/r4618-ir/html/index.htm?goto=2:3:6:1:1:7980

The Validator does not complain about the direct linking, however because ```NetworkFacility``` is not a feature type, deegree does not generate external ```GetFeatureById``` links for referencing the respective objects.

Example WFS: https://haleconnect.com/ows/services/org.874.512633c0-2899-4f9d-b2d1-1bd51cb2315a_wfs?SERVICE=WFS&REQUEST=GetCapabilities&VERSION=2.0.0

**Comment by KathiSchleidt** [comment]
Hi Kate,

to the fact that the validator validates, to my knowledge this doesn't mean much for Annex III themes, as they're not checked against the schema yet, only that the service is OK, and the output is GML

to the issue itself - fear the problem is the automatic conversion of conceptual models to XSD, not taking provision technology into account. From where we stand, you could include the NetworkFacility dataType within the Facility or Network, then provide the link from here. Only "ugly bit" is that this NetworkFacility must provide both links, so you'd have to include a reference to the feature (Network or Facility) where this is included (so a type of backlink).

Problem is that if you don't include this type, there's no way of knowing during which timespan the Facility was linked to the Network, a mandatory attribute in the data model.

**Comment by JohannaOtt** [comment]
Hi all,

> Problem is that if you don't include this type, there's no way of knowing during which timespan the Facility was linked to the Network, a mandatory attribute in the data model.

Just for the sake of completeness: NetworkFacility.linkingTime is voidable - so in case it is not present, delivering a NetworkFacility object would only serve to let the customer know that there is no linkingTime.


The same issue exists for Soil BTW https://inspire.ec.europa.eu/data-model/approved/r4618-ir/html/EARoot/EA2/EA3/EA17/EA1/EA8714.htm where the association between SoilBody and DerivedSoilProfile is using the association class DerivedProfilePresenceInSoilBody which is also containing only one voidable attribute.


All the best

Johanna

**Comment by sMorrone** [comment]
Dear all,
jumping in just to clarify that the INSPIRE Validator has ETS for EF data theme since v2021.0  (18/01/2021). This means that the EF datasets are tested against general as well as theme-specific requirements.
More information on specific tests at https://github.com/inspire-eu-validation/data-ef/tree/master/ef-ia 
Best,
Stefania

**Comment by JohannaOtt** [comment]
Hi all,

I validated [this file](https://haleconnect.com/ows/services/org.874.512633c0-2899-4f9d-b2d1-1bd51cb2315a_wfs?SERVICE=WFS&VERSION=2.0.0&REQUEST=GetFeature&NAMESPACES=xmlns%28ef%2Chttp%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fef%2F4.0%29%2Cxmlns%28ompr%2Chttp%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fompr%2F3.0%29&TYPENAMES=ompr%3AProcess%2Cef%3AEnvironmentalMonitoringNetwork%2Cef%3AEnvironmentalMonitoringFacility) returned by the WFS Kate mentioned in the issue description with the ETS for EF that is now available. In the [result](https://inspire.ec.europa.eu/validator/test-run/details.html?id=EID4a37446b-55f0-4ce8-a282-53cdb549ec9f), there is no error on the NetworkFacility class not being used.
So the validator does not seem to expect it but the schema does?


All the best

Johanna

**Comment by KathiSchleidt** [comment]
The validator does not go into the details of what you link via xlink, you can put anything there, e.g. xlink:href="nonexistant.site.org/dummy"

I did a simple test file for ef:EnvironmentalMonitoringNetwork, added the ef:NetworkFacility, but then dead-ended the ef:contains & ef:belongsTo to point to "#bla" - validates just fine ;) See:
`https://gist.githubusercontent.com/KathiSchleidt/3103e46217d971dd5fa7217e650cf34b/raw/5f486d8707722fc85656cfb515bd108d49b4f36c/EF-XLink-Hack.gml
`

The issue with leaving out the ef:NetworkFacility will be with other applications expecting a correct data model, they'll probably stumble over the gap as they'll be expecting a ef:NetworkFacility where you provide ef:EnvironmentalMonitoringFacility

It comes down to your goal, is this just formal validation or potential reuse of harmonized data?

**Comment by fabiovinci** [comment, marked as answer]
Dear @Kate-Lyndegaard,

the answer to your question "_Is directly linking between EnvironmentalMonitoringNetwork and EnvironmentalMonitoringFacility without using the association class NetworkFacility allowed?_" is "Yes".

The additional attributes of the association `belongsTo`, defined by the association class _NetworkFacility_, are optional.

For this reason, the encoding present in your dataset is valid (as confirmed by Johanna):

`<ef:belongsTo xlink:href="#EnvironmentalMonitoringNetwork_OBS_L2"/>`

Let us know if you have further questions.
